run_pipeline.sh

```diff
@@ -1,72 +1,86 @@
-/FastQC/fastqc data/raw_data/* --outdir data/fastqc_reports
+#----------------------------Make directories---------------------------------#
+mkdir ${dir}
+mkdir ${dir}/raw_data
+mkdir ${dir}/aligned_reads
+mkdir ${dir}/fastqc_reports
+mkdir ${dir}/salmon_quants
+mkdir results
+mkdir ref_files
 
-# Obtain summary report:
-Rscript 1-get_fastqc_reports.R -d data/fastqc_reports -o results
+#--------------Will run setup only if it hasn't been ran before----------------#
+# Will check for genome index first before running
+if [ ! -e ref_files/human_index ]; then
 
-#--------------Paired trimming of the adapters from these data:----------------#
-# Go to the raw_data directory 
-cd data/raw_data
+# Get the human transcriptome
+curl ftp://ftp.ensembl.org/pub/release-94/fasta/homo_sapiens/cdna/Homo_sapiens.GRCh38.cdna.all.fa.gz \
+-o ref_files/Homo_sapiens.GRCh38.cdna.all.fa.gz
 
-# Do adapter trimming for all sample file pairs
-for f in `ls *_1.fastq.gz | sed 's/_1.fastq.gz//'`
-do
-/TrimGalore-0.4.5/trim_galore \
---nextera \
---paired ${f}_1.fastq.gz ${f}_2.fastq.gz \
--o ../fastqc_trimmed
-done
+# Index the human transcriptome
+salmon --threads=16 --no-version-check index \
+-t ref_files/Homo_sapiens.GRCh38.cdna.all.fa.gz \
+-i ref_files/human_index \
+-k 23
+fi
 
-#--------------- Align trimmed reads to genome using HISAT2--------------------#
-# Get premade human genome index:
-wget ftp://ftp.ccb.jhu.edu/pub/infphilo/hisat2/data/grch38.tar.gz -P ../
-tar data/grch38.tar.gz
-
-# Go to trimmed reads directory
-cd ../fastqc_trimmed
+#--------------------------- Download fastq data-------------------------------#
+# Note: because running all ~3800 samples takes quite a bit of time, use -n option 
+# to set what number of randomly selected samples you would like  
+Rscript scripts/0-get_sample_download_list.R \
+-i ${SRP} \
+-o results \
+-d ${dir}/salmon_quants \
+-q ref_files/SRAmetadb.sqlite \
 
-# Do genome alignment for all sample pairs
-for f in `ls -1 *_1_val_1.fq.gz | sed 's/_1_val_1.fq.gz//' `
-do
-/hisat2-2.1.0/hisat2 -x ../grch38/genome \
--1 ${f}_1_val_1.fq.gz \
--2 ${f}_2_val_2.fq.gz \
--S ../aligned_reads/${f}.bam
-done
+# Change directory to the data
+cd ${dir}
 
-#--------------------Further prep the bam files with samtools------------------#
-# Change to directory where bam files are
-cd ../aligned_reads
-# Sort and index bam files
-for f in `ls *.bam | sed 's/.bam//' `
+# Download each sample and run Salmon on it. Then remove the sample to save room. 
+for line in `cat ../files.to.download.txt`
 do
-/samtools-1.3.1/samtools sort ${f}.bam -o ${f}.sorted.bam
-/samtools-1.3.1/samtools index ${f}.bam
+  # Download forward and reverse fastq files
+  Rscript ../scripts/1-download_sra.R \
+  -s $line \
+  -q ../ref_files/SRAmetadb.sqlite \
+  -d raw_data
+  
+  # Run sequence quality control with FASTQC
+  /FastQC/fastqc raw_data/* --outdir fastqc_reports
+  
+  # For each fastq file pair, do QC and then salmon
```

Comment:
Doing QC in line 58?

Reply:
Yes. Should I not?

Reply:
Which part is confusing?

Reply:
You're not doing any QC in the loop, correct?

Reply:
I just got what you mean.